app/routes/_gcn.circulars/circulars.server.ts

```diff
@@ -94,41 +94,58 @@ export async function search({
   limit,
   startDate,
   endDate,
+  useNLP
 }: {
   query?: string
   page?: number
   limit?: number
   startDate?: string
   endDate?: string
+  useNLP?: boolean
 }): Promise<{
   items: CircularMetadata[]
   totalPages: number
   totalItems: number
 }> {
   const client = await getSearch()
-
   const [startTime, endTime] = getValidDates(startDate, endDate)
 
-  const {
-    body: {
-      hits: {
-        total: { value: totalItems },
-        hits,
-      },
-    },
-  } = await client.search({
-    index: 'circulars',
-    body: {
-      query: {
+  const get_model_id_request = {
+    method: 'GET',
+    path: '/_ingest/pipeline',
+  }
+
+  let model_id = ''
+  try {
+    const resp = await client.transport.request(get_model_id_request)
+
+    if (resp && resp.statusCode == 200) {
+      model_id =
+        resp.body['nlp-ingest-pipeline'].processors[0].text_embedding.model_id
+    } else {
+      console.log(
+        'Error. Could not deploy model. Returned with response: ',
+        resp
+      )
+    }
+  } catch (e) {
+    console.log('Error: ', e)
+  }
```

Comment:
Would you please see if you can use the [raw HTTP request API](https://github.com/opensearch-project/opensearch-js/blob/main/guides/json.md) that was recently added to the OpenSearch client library?

Reply:
Sure.